test/distributed/fsdp/test_fsdp_summon_full_params.py

```diff
@@ -0,0 +1,278 @@
+# Owner(s): ["oncall: distributed"]
+import sys
+import math
+
+import torch
+import torch.nn as nn
+from torch import distributed as dist
+from torch.distributed.fsdp import FullyShardedDataParallel as FSDP
+from torch.distributed.fsdp import CPUOffload
+from torch.testing._internal.common_distributed import skip_if_lt_x_gpu
+from torch.testing._internal.common_fsdp import (
+    FSDPTest,
+)
+from torch.testing._internal.common_utils import (
+    TEST_WITH_DEV_DBG_ASAN,
+    run_tests,
+    instantiate_parametrized_tests,
+    parametrize,
+)
+
+
+if not dist.is_available():
+    print("Distributed not available, skipping tests", file=sys.stderr)
+    sys.exit(0)
+
+if TEST_WITH_DEV_DBG_ASAN:
+    print(
+        "Skip dev-asan as torch + multiprocessing spawn have known issues",
+        file=sys.stderr,
+    )
+    sys.exit(0)
+
+
+class TestSummonFullParams(FSDPTest):
+    @property
+    def world_size(self):
+        return 2
+
+    def get_model_param_count(self, m):
+        return sum([p.numel() for p in m.parameters()])
+
+    # padding ensures that all shards have the same size with the least amount of padding
+    def get_expected_sharded_size(self, global_size):
+        return int(math.ceil(global_size / self.world_size))
+
+    @skip_if_lt_x_gpu(2)
+    @parametrize(
+        "writeback",
+        [True, False]
+    )
+    @parametrize(
+        "cpu_offload",
+        [CPUOffload(offload_params=True), CPUOffload(offload_params=False)]
+    )
+    @parametrize(
+        "modify_outer",
+        [True, False]
+    )
+    def test_summon_full_param_writeback(self, writeback, cpu_offload, modify_outer):
+        model = FSDP(nn.Sequential(
+            FSDP(nn.Linear(5, 5, bias=False)),
+            nn.Linear(5, 3, bias=False)
+        )).cuda(self.rank)
```

Comment:
Missed this in code review, but it does not actually respect the cpu_offload argument.